package tests;

import org.junit.jupiter.api.AfterEach;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.chrome.ChromeOptions;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class ExampleTest{
    WebDriver driver;
    @BeforeEach
    public void setUp(){
        ChromeOptions options = new ChromeOptions();

        //importing a headless jenkins
        options.addArguments("--headless=new");
        options.addArguments("--window-size=1920,1980");

        driver = new ChromeDriver(options);
    }

    @Test
    public void verifyExampleWebsiteTitle() {
        driver.get("https://example.com");
        String actualTitle = driver.getTitle();
        System.out.println("Page Title = " + actualTitle);
        assertEquals("Example Domain", actualTitle);
    }

    @AfterEach
    public void tearDown(){
        if(driver != null){
            driver.quit();
        }
    }
}